src/mongo_client.ts

```diff
@@ -346,19 +346,29 @@ export class MongoClient extends EventEmitter implements OperationParent {
     const force = typeof forceOrCallback === 'boolean' ? forceOrCallback : false;
 
     return maybePromise(callback, cb => {
+      const completeClose = (err?: AnyError) => {
```

Comment:
nit: naming of `completeClose`, perhaps something like `resetState` is more apt?

Reply:
Had a small offline discussion. Leaning towards leaving this as-is because `completeClose` both resets the state and calls the callback. Does anyone else have thoughts on this?

Reply:
I'm fine with `completeClose`, it's also a pattern we use elsewhere